src/HDF5.jl

```diff
@@ -1378,26 +1376,41 @@ function getindex(parent::Union{HDF5File, HDF5Group, HDF5Dataset}, r::HDF5Refere
     h5object(obj_id, parent)
 end
 
+# Helper for reading compound types
+function read_row(io::IO, membertype, membersize)
+    row = Any[]
+    for (dtype, dsize) in zip(membertype, membersize)
+        if dtype === String
+            push!(row, strip(String(read(io, UInt8, dsize)),'\0'))
```

Comment:
There is a method `unpad` defined elsewhere in this file that deals with different string padding modes. Can we use that in combination with `h5t_get_strpad` to handle more general cases?